packages/dds/shared-object-base/src/sharedObjectBase.ts

```diff
@@ -0,0 +1,448 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { v4 as uuid } from "uuid";
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { assert } from "@fluidframework/common-utils";
+import { AttachState } from "@fluidframework/container-definitions";
+import { IFluidHandle, IFluidSerializer } from "@fluidframework/core-interfaces";
+import {
+    IChannelAttributes,
+    IFluidDataStoreRuntime,
+    IChannelStorageService,
+    IChannelServices,
+    IStateHost,
+} from "@fluidframework/datastore-definitions";
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import {
+    IGarbageCollectionData,
+    ISummaryTreeWithStats,
+} from "@fluidframework/runtime-definitions";
+import { FluidSerializer } from "@fluidframework/runtime-utils";
+import { ChildLogger, EventEmitterWithErrorHandling } from "@fluidframework/telemetry-utils";
+import { SharedObjectHandle } from "./handle";
+import { SummarySerializer } from "./summarySerializer";
+import { ISharedObject, ISharedObjectEvents } from "./types";
+
+export interface ISharedObjectStateHost {
```

Comment:
I'm not sure what "host" refers to. I think ISharedObjectSnapshot would be appropriate name, but "snapshot" is so overloaded term that I do not want to use it here. Also, I think long term we want to extend that beyond DDS / SharedObject (base class). Looking for some name / synonyms that captures intent - IObjectRevision or IRevision ?